independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/handlers/ResourceLocatorHandler.java

```diff
@@ -80,7 +80,7 @@ public void onComplete(Throwable throwable) {
         requestContext.setRemaining(res.remaining);
         requestContext.setEndpointInstance(locator);
         requestContext.setResult(null);
-        requestContext.restart(res.value);
+        requestContext.restart(res.value, true);
```

Comment:
Is there a reason to overwrite the `pathParamValue` for locators? Is it because they can declare new path parameters? Otherwise we can just keep the original `pathParamValue` and no need to add a field to save them.

Reply:
The subresource can indeed declare different parameters - they don't need to be same as the original. So to support that case we do indeed need to override

Reply:
OK